src/main/java/stirling/software/SPDF/controller/api/misc/ExtractImagesController.java

```diff
@@ -145,75 +231,116 @@ private boolean shouldUseMultithreading(MultipartFile file, PDDocument document)
 
     private void extractImagesFromPage(
             PDPage page,
+            int pageNum, // Add pageNum parameter
             String format,
             String filename,
-            int pageNum,
+            AtomicInteger imageIndex,
             Set<byte[]> processedImages,
             ZipOutputStream zos,
             boolean allowDuplicates)
             throws IOException {
+
         MessageDigest md;
         try {
             md = MessageDigest.getInstance("MD5");
         } catch (NoSuchAlgorithmException e) {
-            logger.error("MD5 algorithm not available for extractImages hash.", e);
+            logger.error("MD5 algorithm not available for image hashing.", e);
             return;
         }
+
         if (page.getResources() == null || page.getResources().getXObjectNames() == null) {
             return;
         }
+
         int count = 1;
         for (COSName name : page.getResources().getXObjectNames()) {
             if (page.getResources().isImageXObject(name)) {
                 PDImageXObject image = (PDImageXObject) page.getResources().getXObject(name);
-                if (!allowDuplicates) {
-                    byte[] data = ImageProcessingUtils.getImageData(image.getImage());
-                    byte[] imageHash = md.digest(data);
-                    synchronized (processedImages) {
-                        if (processedImages.stream()
-                                .anyMatch(hash -> Arrays.equals(hash, imageHash))) {
+                try {
+                    if (!allowDuplicates) {
```

Comment:
The duplicate logic doesnt seem to work anymore looking at the github PR runs
shows that the extract test extracts 5 images instead of 2 (when 3 are dups and should be removed)

Reply:
You can test with https://github.com/Stirling-Tools/Stirling-PDF/blob/main/cucumber/exampleFiles/images.pdf

Reply:
sure. I will check, test and update